pkg/storage/bucket/s3/bucket_client.go

```diff
@@ -153,8 +154,18 @@ func (b *BucketWithRetries) Exists(ctx context.Context, name string) (exists boo
 }
 
 func (b *BucketWithRetries) Upload(ctx context.Context, name string, r io.Reader) error {
+	// Convert Reader to Seeker
+	var buf bytes.Buffer
+	if _, err := buf.ReadFrom(r); err != nil {
```

Comment:
Would it be possible to minimized the amount we have to cache everything into memory? I am thinking about a case where the files we are uploading is huge, and if put them into memory it might add to memory overhead.

Reply:
opps, @alanprot bet me to it :)